### PR TITLE
dotCMS/core#20295 : Files with blank spaces in their names and/or mis…

### LINKED ISSUE #20295: [Static Push] : File names with blank spaces cause problems
Reported via Support: [104010](https://dotcms.zendesk.com/agent/tickets/104010)
Reproducible in dotCMS 5.2.8.4, 5.3.8.4, and latest master.

Static pushing a File Asset with blank spaces in its name to an S3 AWS Environment is causing a different behavior compared to pushing a file **WITHOUT** any blank spaces.

**To Reproduce**
**NOTE: If the Demo Starter is not enough to reproduce this, please ask Jose Castro for details on the required data set.**
1. Create a test folder named `master-folder`, and upload a text file named `dotcms-test-file.txt` to it.
2. Set up an AWS S3 environment. You can ask Jose Castro for details on this as well.
3. Push the `dotcms-test-file.txt`. This works as expected. If you check the `dotcms.log`, you'll see this too:
```
INFO  bundlers.FileAssetBundler - +structuretype:4 +(path:/master-folder/dotcms-test-file.txt ) +(conhost:48190c8c-42c4-46af-8d1a-0cd5db894797  ) 
```
This is the Lucene query used to pull the file you selected.

4. Now, under the same `master-folder`, upload another text file named `dotcms test file`. Notice that the file has blank spaces, and no file extension.
5. Push this new file to S3. The Lucene looks like this now:
```
INFO  bundlers.FileAssetBundler - +structuretype:4 +(path:/master-folder/dotcms test file ) +(conhost:48190c8c-42c4-46af-8d1a-0cd5db894797  ) 
```
This query matches the file we selected, along with many other unrelated files in the repository. This is wrong.

6. Additionally, the push status is marked as `Successful` even though some of the pulled files actually failed to push.

**UPDATE on Apr 26th: Passing down a file with blank spaces is also causing the `HTMLPageAsContentBundler` to run and include a very high number of completely unrelated HTML Pages in the push.**

**Expected behavior**
Files with or without blank spaces in their names must be able to be static pushed correctly.

**Additional context**
In some customer environments, depending on the amount and size of files that are being pulled incorrectly, this is causing severe problems with the publishing queue

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…sing file extension are causing problems when static pushing them to AWS S3. The Lucene property that takes the path to the file was taking each word of the file name as a separate search criterion. This was making dotCMS pull additional and unrelated information during the push.

Bundlers in Static Publishing are called with file(s) you're pushing via the FileAssetBundler. However, the same file with blank spaces is used to find potential HTML Pages, and URL Maps. This was causing problems those other bundlers as well.

 Adding missing Javadoc, refactoring, and improving several error messages to include more information.

### CODE REVIEW

**Comment by erickgonzalez** [comment]
@jcastro-dotcms do you think it's possible to do some tests here?

**Comment by dsilvam** [comment]
Same request as @erickgonzalez

File: .gitmodules [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [submodule "dotCMS/src/main/enterprise"]
 	path = dotCMS/src/main/enterprise
 	url = git@github.com:dotCMS/enterprise.git
-	branch = master
+	branch = issue-20295-filenames-with-blank-spaces-cause-problems
```
Comment (jdotcms) [inline .gitmodules:4]: probably need to discard this change

File: dotCMS/dependencies.gradle [hunk 2]
```diff
@@ -3,7 +3,10 @@ dependencies {
 
     def eeType = dotcmsReleaseVersion.equals("master") ? '-SNAPSHOT' : ''
 
-    compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    //compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    //compile fileTree(dir: 'src/main/enterprise/build/libs', include: ['ee_clean.jar'])
+    compile group: 'com.dotcms.enterprise', name: 'ee', version: 'issue-20295-filenames-with-blank-spaces-cause-problems-20210504.041230-2', changing: true
```
Comment (jdotcms) [inline dotCMS/dependencies.gradle:8]: same here

File: dotCMS/gradle.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-dotcmsReleaseVersion=master
+dotcmsReleaseVersion=issue-20295-filenames-with-blank-spaces-cause-problems
```
Comment (jdotcms) [inline dotCMS/gradle.properties:1]: same here

**Review by nollymar — APPROVED** [review]

File: dotCMS/gradle.properties (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/gradle.properties:1]: I wish there was an easier way to force a PR to run against a specific version of enterprise without having to make all these changes. Then see how people think you "probably" added them by accident and request you get them removed. 
Yes. they need to be removed.

**Review by jdotcms — APPROVED** [review]

**Pull request merged**